### PR TITLE
Fix memory leaks

### PR DESCRIPTION
**Pull request by dmshynk**
Got started on fixing various memory leaks possibly causing  #2308 and #2217. Several issues still exist, working on them.

### CODE REVIEW

File: src/widget/systemtrayicon.cpp [hunk 1]
```diff
@@ -114,6 +114,8 @@ SystemTrayIcon::SystemTrayIcon()
                                  8, image->width(), image->height(),
                                  image->bytesPerLine(), callbackFreeImage, image);
 
+        delete image;   // Not needed anymore.
```
Comment (antis81) [inline src/widget/systemtrayicon.cpp]: Why not change `QImage*` to `QImage`? Also solves the ugly `*image = …` in line 111 above.

Comment (agilob) [inline src/widget/systemtrayicon.cpp]: @antis81 merge this when you think it's ready.

**Comment by antis81** [comment]
@dmshynk Ready for merge? I have some suggestions here to make life easier, but those can be discussed in another PR, I think.

**Comment by dmshynk** [comment]
All ready. I'm now struggling with the next leak, which seems to be out of my control (see my comment to #2217), so any suggestions are welcome.

**Comment by antis81** [comment]
@dmshynk :+1:

File: src/widget/genericchatroomwidget.cpp [hunk 2]
```diff
@@ -72,6 +72,7 @@ void GenericChatroomWidget::compactChange(bool _compact)
     // avatar
     if (isCompact())
     {
+        delete textLayout;  // Not needed
```
Comment (antis81) [inline src/widget/genericchatroomwidget.cpp:75]: You can move recreation of `textLayout` (c'tor in line 63) here and replace the delete - this is safe. As I'm not sure what that nasty comment on the above delete (line 59) means I'd leave that untouched for now.

**Comment by dmshynk** [comment]
@antis81, thanks for the valuable feedback, will do tomorrow morning. I am sort of anxious to mess with working code too much when it comes to memory management issues, but this indeed seems to be safe.

File: src/widget/genericchatroomwidget.cpp (hunk 2 above)
Comment (antis81) [inline src/widget/genericchatroomwidget.cpp:75]: Argh no, my proposed change will lead to crash (delete dangling pointer on subsequent calls). You would also need to set `textLayout = nullptr;` after line 59. Anyway, I'll happily help to clean this up in a new PR - merging this one later today.

**Pull request merged**